nsqd/options.go

```diff
@@ -28,6 +28,12 @@ type nsqdOptions struct {
 	SyncEvery       int64         `flag:"sync-every"`
 	SyncTimeout     time.Duration `flag:"sync-timeout"`
 
+	QueueGCInterval        time.Duration
```

Comment:
I'm not a fan of `GC` here, but i don't have much of a better suggestion. Naming here feels complicated because this controls both the in-flight timeouts, and the deferred queue timeouts. If we were to only describe this in terms of Deferred or in-flight it might be easier?

`DeferredQueueCheckInterval`, `DeferredQueueScalingInterval` (i'm not sure we should even expose this one), `DeferredQueueMaxWorkers` (or max concurrent?), `DeferredQueueDirtyPercent`. 

The other naming pattern that comes to mind would be `RequeueCheckInterval`, ....